### PR TITLE
Settlement report refactor

### PR DESCRIPTION
**Pull request by 00eadeyemi**
### Features 

- Refactoring out aggregate settlement reports. 

[Delivers #130356905](https://www.pivotaltracker.com/story/show/)

### CODE REVIEW

**Comment by 00eadeyemi** [comment]
@srleach please pre-review

File: app/Http/Controllers/SettlementsController.php [hunk 1]
```diff
@@ -98,19 +98,23 @@ public function settlementReport($merchant, $id)
             $settlementReport = $this
                 ->settlementGateway
                 ->getSingleSettlementReport($this->fetchMerchantById($merchant)->token, $id);
+
+            $aggregateSettlementReport = $this
+                ->settlementGateway
+                ->getSingleAggregateSettlementReport($this->fetchMerchantById($merchant)->token, $id);
+
+            $aggregateSettlementReport['sum_net'] = array_sum(array_column($aggregateSettlementReport, 'Settlement Amount'));
```
Comment (srleach) [inline app/Http/Controllers/SettlementsController.php]: Readability

File: app/Http/Controllers/SettlementsController.php [hunk 2]
```diff
@@ -98,19 +98,23 @@ public function settlementReport($merchant, $id)
...
         } catch (\Exception $e) {
             throw $this->redirectWithException('/', 'Failed fetching settlements', $e);
         }
 
-        $this->applySettlementAmounts($settlementReport);
-
         return View('settlements.settlement_report', [
             'settlementReport' => $settlementReport,
+            'aggregateSettlementReport' => $aggregateSettlementReport,
```
Comment (srleach) [inline app/Http/Controllers/SettlementsController.php]: Naming Standards inconsistent

File: resources/views/settlements/settlement_report.blade.php [hunk 3]
```diff
@@ -5,7 +5,7 @@
     <h1>
         Settlement Report
         <div class="btn-toolbar pull-right">
-            <a href="{{ Request::url() }}/?download=csv&amp;source=view_data&amp;filename={{ $export_view_filename }}" class="btn btn-default"><span class="glyphicon glyphicon-save"></span> Download Report</a>
+            <a href="{{ Request::url() }}/?download=csv&amp;source=aggregateSettlementReport&amp;filename={{ $export_view_filename }}" class="btn btn-default"><span class="glyphicon glyphicon-save"></span> Download Report</a>
```
Comment (srleach) [inline resources/views/settlements/settlement_report.blade.php]: changing standards?

**Comment by 00eadeyemi** [comment]
@srleach

File: resources/views/settlements/settlement_report.blade.php [hunk 4]
```diff
@@ -41,26 +41,26 @@
...
-                        <td class="text-right">{{ '&pound;' . number_format($item['order_amount']/100, 2)}}</td>
-                        <td>{{ $item['type'] }}</td>
-                        <td class="{{($item['deposit'] < 0 ? 'text-danger' : '') }} text-right">{{'&pound;' . number_format($item['deposit']/100, 2)}}</td>
-                        <td class="{{($item['loan_amount'] < 0 ? 'text-danger' : '') }} text-right" >{{'&pound;' . number_format($item['loan_amount']/100, 2)}}</td>
-                        <td class="{{($item['subsidy'] < 0 ? 'text-danger' : '') }} text-right">{{'&pound;' . number_format($item['subsidy']/100, 2) }}</td>
-                        <td class="{{($item['adjustment'] < 0 ? 'text-danger' : '') }} text-right">{{'&pound;' . number_format($item['adjustment']/100, 2)}}</td>
-                        <td class="{{($item['net'] < 0 ? 'text-danger' : '') }} text-right">{{'&pound;' . number_format($item['net']/100, 2) }}</td>
+                        <td>{{ date('d/m/Y', strtotime($item['Order Date'])) }}</td>
+                        <td>{{ date('d/m/Y', strtotime($item['Notification Date'])) }}</td>
```
Comment (srleach) [inline resources/views/settlements/settlement_report.blade.php]: Are these array keys?

Comment (srleach) [inline resources/views/settlements/settlement_report.blade.php]: Why have we changed the standard here?

**Comment by srleach** [comment]
@00eadeyemi

File: resources/views/settlements/settlement_report.blade.php (hunk 4 above)
Comment (wnowicki) [inline resources/views/settlements/settlement_report.blade.php]: seriously... is that not on API ?

**Pull request merged**